ui/src/kapacitor/components/config/KafkaConfigs.tsx

```diff
@@ -0,0 +1,120 @@
+import React, {Component, MouseEvent} from 'react'
+import _ from 'lodash'
+
+import KafkaConfig from 'src/kapacitor/components/config/KafkaConfig'
+import {ErrorHandling} from 'src/shared/decorators/errors'
+
+import {KafkaProperties} from 'src/types/kapacitor'
+import {Notification, NotificationFunc} from 'src/types'
+
+import {get} from 'src/utils/wrappers'
+
+interface Config {
+  options: KafkaProperties & {
+    id: string
+  }
+  isNewConfig?: boolean
+}
+
+interface Props {
+  configs: Config[]
+  onSave: (properties: KafkaProperties) => void
+  onDelete: (specificConfig: string) => void
+  onTest: (
+    e: MouseEvent<HTMLButtonElement>,
+    specificConfigOptions: Partial<KafkaProperties> & {id: string}
```

Comment:
could an `Exclude` be used here, in case we know which are required? and just make the ones conditional that are?

Reply:
Not exactly, everything is optional except for id so this is the most concise way to convey that.